services/src/main/java/org/keycloak/authentication/AuthenticationSelectionResolver.java

```diff
@@ -193,7 +193,7 @@ private static boolean addAllExecutionsFromSubflow(AuthenticationProcessor proce
 
                 // For conditional execution, we must check if condition is true. Otherwise return false, which means trying next
                 // requiredExecution in the list
-                return !flow.isConditionalSubflowDisabled(ex);
+                return !flow.isConditionalSubflowDisabled(ex, false);
```

Comment:
Could you please elaborate on why not track the matching result when evaluating conditional authenticators?

Reply:
When the selection page for authenticators is shown, the results must not be cached because the outcome of the condition may be different during the authenticaion flow process. That is because the level of authentication changes during the flow and must be reevaluated in the conditions.

Reply:
> It' true that a callback method would make the implementation simpler. As this would be a change to the core ConditionalAuthenticator interface and its processing, we would prefer it, if this could be implemented by the main keycloak developers.

I'm good either way. Let's see what @mposolda thinks about it.